test/DynamoCoreWpfTests/PackageManager/PackageManagerSearchElementViewModelTests.cs

```diff
@@ -1,50 +1,97 @@
+            var name2 = "duplicate";
+            var version = "1.0.0";
+
+            var mockGreg = new Mock<IGregClient>();
+            var clientmock = new Mock<PackageManagerClient>(mockGreg.Object, MockMaker.Empty<IPackageUploadBuilder>(), string.Empty);
+            var pmCVM = new Mock<PackageManagerClientViewModel>(ViewModel, clientmock.Object);
+
+            var ext = Model.GetPackageManagerExtension();
+            var loader = ext.PackageLoader;
+            loader.Add(new Package("", name2, version, ""));
+
+            var packageManagerSearchViewModel = new PackageManagerSearchViewModel(pmCVM.Object);
+            packageManagerSearchViewModel.RegisterTransientHandlers();
+
+            var newSE1 = new PackageManagerSearchElementViewModel(new PackageManagerSearchElement(new PackageHeader()
+            {
+                name = name1
+            }), false);
+
+            var newSE2 = new PackageManagerSearchElementViewModel(new PackageManagerSearchElement(new PackageHeader()
+            {
+                name = name2
+            }), false);
+
+            packageManagerSearchViewModel.AddToSearchResults(newSE1);
+            packageManagerSearchViewModel.AddToSearchResults(newSE2);
+
+            // Default CanInstall should be true
+            Assert.AreEqual(true, newSE1.CanInstall);
+            Assert.AreEqual(true, newSE2.CanInstall);
+
+            var dHandle1 = new PackageDownloadHandle()
+            {
+                Id = name1,
+                VersionName = version,
+                Name = name1
+            };
+
+            var dHandle2 = new PackageDownloadHandle()
+            {
+                Id = name2,
+                VersionName = version,
+                Name = name2
+            };
+
+            pmCVM.Object.Downloads.Add(dHandle1);
+            pmCVM.Object.Downloads.Add(dHandle2);
+
+            Assert.AreEqual(true, newSE1.CanInstall);
+            Assert.AreEqual(true, newSE2.CanInstall);
+
+            dHandle1.DownloadState = PackageDownloadHandle.State.Downloading;
+            Assert.AreEqual(false, newSE1.CanInstall);
+
+            dHandle1.DownloadState = PackageDownloadHandle.State.Downloaded;
+            Assert.AreEqual(false, newSE1.CanInstall);
+
+            dHandle1.DownloadState = PackageDownloadHandle.State.Error;
+            dHandle2.DownloadState = PackageDownloadHandle.State.Downloading;
+
+            Assert.AreEqual(true, newSE1.CanInstall);
+            Assert.AreEqual(false, newSE2.CanInstall);
+
+            dHandle1.DownloadState = PackageDownloadHandle.State.Installing;
+            dHandle2.DownloadState = PackageDownloadHandle.State.Downloaded;
+            Assert.AreEqual(false, newSE1.CanInstall);
+            Assert.AreEqual(false, newSE2.CanInstall);
+
+            dHandle1.DownloadState = PackageDownloadHandle.State.Installed;
+            dHandle2.DownloadState = PackageDownloadHandle.State.Installed;
+            Assert.AreEqual(true, newSE1.CanInstall);
```

Comment:
How come this is true, the package is installed, right?

Reply:
In this test scenario (and I can add some more descriptions) there are 2 packages being downloaded:
            var name1 = "non-duplicate"; (downloadHandle newSE1)
            var name2 = "duplicate"; (downloadHandle newSE2)
newSE1 - has no conflicts with existing localPackages
newSE2 - has a conflict with localPackcages  
          
Moreover newSE1 never gets added to localPackages (that's just what I chose to do in the test). Because of this, after newSE1 gets an  Installed downloadState, it can still be installed. I guess this could simulate an exception being thrown before the newSE1 package can get loaded in localPackages

Reply:
So it looks like this can never be a real-life scenario. If so, then why have a test for it?

Reply:
I added the newSE1 to local packages in the end (simulating a successful load by the packageLoader).
Now both packages cannot be installed